velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -407,14 +407,33 @@ struct ArrayFrequencyFunction {
   FOLLY_ALWAYS_INLINE void call(
       out_type<velox::Map<T, int>>& out,
       arg_type<velox::Array<T>> inputArray) {
-    folly::F14FastMap<arg_type<T>, int> frequencyCount;
+    frequencyCount_.clear();
 
     for (const auto& item : inputArray.skipNulls()) {
-      frequencyCount[item]++;
+      frequencyCount_[item]++;
     }
 
-    out.copy_from(frequencyCount);
+    // To make the output order of key value pairs deterministic (since F14
+    // does not provide ordering guarantees), we do another iteration in the
+    // input and look up element frequencies in the F14 map. To prevent
+    // duplicates in the output, we replace the frequency count with a -1.
```

Comment:
Not sure whether we have concern about this overhead. Should we consider leaving it non-deterministic and enhancing fuzzer to not verify the results of non-deterministic functions?

Reply:
We can add a small benchmark to test if we feel like this is important, but my impression is that another sequential pass over this array should add very little overhead. Also, considering that we were allocating the F14 map per record on the previous version, this version would still be much faster regardless (since now we don't allocate as much). 

My main concern about making it non-deterministic is that it deviates from PrestoSQL (array_frequency is deterministic there), but I'm open to suggestions. 

@mbasmanova any thoughts on this?